### PR TITLE
throw error only if multiple reporters really write to same stream (closes #4675)

### LINKED ISSUE #4675: TestCafe always forces you to set output file for second reporter
TestCafe forces you to set an output file for a second reporter even if it doesn't output to stdout
(e.g. the dashboard reporter outputs neither to file nor to stdout.)

### What is your Test Scenario?
I need to use spec reporter with dashboard reporter for TestCafe tests:
`testcafe chrome -r spec, dashboard`
 
### What is the Current behavior?
TestCafe doesn't start tests and throws the following error: `Multiple reporters attempting to write to stdout: "spec, dashboard". Only one reporter can write to stdout.`
 
### What is the Expected behavior?
TestCafe start tests without error.

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

File: test/functional/fixtures/regression/gh-4675/test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+            return createTestCafe('127.0.0.1', 1335, 1336)
+                .then(tc => {
+                    testCafe = tc;
+                })
+                .then(() => {
+                    return testCafe.createRunner()
+                        .browsers(`chrome`)
+                        .src(path.join(__dirname, './testcafe-fixtures/index.js'))
+                        .reporter([customReporter, customReporter])
```
Comment (miherlosev) [inline test/functional/fixtures/regression/gh-4675/test.js]: Can specify a different reporter names to scenario 'All target reporter names are present in the error message'?

File: src/runner/index.js [hunk 2]
```diff
@@ -169,6 +170,22 @@ export default class Runner extends EventEmitter {
             completed = true;
         };
 
+        let multipleReportersError = null;
+
+        reporterStreamController.once('multiple-reporters-error', error => {
```
[the remark this answers is not in the document]
Comment (AlexKamaev) [reply src/runner/index.js]: I refactored this code and removed this message

File: src/runner/index.js [hunk 3]
```diff
@@ -169,6 +170,22 @@ export default class Runner extends EventEmitter {
...
+            multipleReportersError = new GeneralError(RUNTIME_ERRORS.multipleStdoutReporters, error);
```
[the remark this answers is not in the document]
Comment (AlexKamaev) [reply src/runner/index.js]: @VasilyStrelyaev need your help

File: test/functional/fixtures/regression/gh-4675/test.js (hunk 1 above)
Comment (AlexKamaev) [reply test/functional/fixtures/regression/gh-4675/test.js]: Don't get it. You can specify different reporter names. Reporter's name mean nothing, only 'stream'

File: test/functional/fixtures/regression/gh-4675/test.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                        .run();
+                })
+                .catch(err => {
+                    error = err;
+
+                    return testCafe.close();
+                })
+                .finally(() => {
+                    expect(error.message).eql('Multiple reporters attempting to write to stdout: "custom, custom". Only one reporter can write to stdout.');
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/regression/gh-4675/test.js]: *The following reporters attempted to write to the same output stream: "custom", "custom". Only one reporter can write to a stream.*

checked by Dirk

File: src/runner/reporter-stream-controller.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        const pluginInfo = this._pluginInfos.find(info => info.stream === stream);
+
+        if (!pluginInfo)
+            this._pluginInfos.push({ stream: stream, plugin: plugin });
+        else if (pluginInfo.plugin !== plugin) {
+            const message = [pluginInfo.plugin.name, plugin.name].join(', ');
+
+            this.multipleStreamError = new GeneralError(RUNTIME_ERRORS.multipleSameStreamReporters, message);
+
+            this._task.emit('done');
```
Comment (AndreyBelym) [inline src/runner/reporter-stream-controller.ts]: Can't we use `task.cancel` here?

Comment (AlexKamaev) [reply src/runner/reporter-stream-controller.ts]: Task has `abort` method, but not `cancel`. The `abort` method does not dispose assets and does not resolve promises inside of reporters.

Comment (AndreyBelym) [reply src/runner/reporter-stream-controller.ts]: Ok then

**Pull request merged**